api/src/main/java/jakarta/persistence/EntityManager.java

```diff
@@ -1054,6 +1054,24 @@ void refresh(Object entity,
      */
     <T> TypedQuery<T> createQuery(String qlString, Class<T> resultClass);
 
+    /**
+     * Create an instance of {@link TypedQuery} for executing a
+     * Jakarta Persistence query language statement, specifying
+     * an {@link EntityGraph} which is interpreted as a load
+     * graph. The select list of the query must contain only a
+     * single item, which must be assignable to the root type
+     * of the given entity graph.
+     * @param qlString a Jakarta Persistence query string
+     * @param resultGraph entity graph interpreted as a load graph
+     * @return the new query instance
+     * @throws IllegalArgumentException if the query string is
+     *         found to be invalid or if the query result is
+     *         found to not be assignable to the root type of
+     *         the entity graph
+     * @since 4.0
+     */
+    <T> TypedQuery<T> createQuery(String qlString, EntityGraph<T> resultGraph);
```

Comment:
Perhaps this should be:

```suggestion
    <T> TypedQuery<? extends T> createQuery(String qlString, EntityGraph<T> resultGraph);
```

since we accept `EntityGraph`s for a supertype of the query result type.